### DISCUSSION TITLE
Simple string values of parser rules seem to not be given ast/cst nodes

### DISCUSSION THREAD

**Discussion by Greglar4** [eclipse-langium/langium#1723] [Q&A] [answered]
Hello,

I am using Langium, and I am currently working on applying edits from a table editor that I am creating to the DSL.

In doing so, I ran into an issue in certain cases. Lets say I have the following parser rule:

Person: 'person' name=STRING

After creating a few people in a file and inspecting the parse results, it seems that the AST nodes for the people have a name property that is simply a string, not an AST node. Thus there seems to be no CST node for the people's names either. I also could not find a property on the CST nodes for the people that would tell me where in the text document the names of the people are.

Is there a way to get the text file location of simple string properties of parser rules?

Thank you,

Greg

**Comment by cdietrich** [comment]
Why do you expect name to be an ast node. STRING is a terminal rule and as a such produces a value

**Comment by Greglar4** [comment]
I guess I do not expect it to be an AST node, I am just looking for a way
to locate where that string is in the text file so that I can edit it

**Reply by cdietrich** [reply, marked as answer]
Maybe GrammarUtil findNodeForProperty helps

**Reply by Greglar4** [reply]
Thanks so much! This seems to be exactly what I was looking for
